delectable-client/src/app/component/recipe-form/recipe-form.component.html

```diff
@@ -1,99 +1,54 @@
-             <button class="btn-light" (click)=removeDirection(direction)>Remove</button>
-          </div>
-       </div>
-    </div>
-    <button class="btn-light" (click)=addDirection()>Add step</button>
-    <h3>Ingredients</h3>
-    <div class="form-group" *ngFor="let ingredient of getIngredientsFormControls.controls; let i = index">
-       <div class="row" [formGroup]="ingredient">
-          <div class="col-auto">
-             <label for="pantryItem">Pantry Item</label>
-             <select class="form-control" id="pantryItem">
-                <option *ngFor="let pantry of PantryItems" value="{{pantry.name}}">{{pantry.name}}</option>
-             </select>
-          </div>
-          <div class="col-auto">
-             <label for="quantity_{{i}}">Quantity</label>
-             <input type="text" class="form-control" id="quantity_{{i}}" formControlName="quantity">
-          </div>
-          <div class="col-auto">
-             <label for="serving_{{i}}">Serving Type</label>
-             <input type="text" class="form-control" id="serving_{{i}}" idformControlName="servingType">
-          </div>
-          <div class="col-auto">
-             <button class="btn-light" (click)=removeIngredient(i)>Remove</button>
-          </div>
-       </div>
-    </div>
-    <button class="btn-light" (click)=addIngredient()>Add ingredient</button>
-    <button class="btn-light" type="submit">Submit</button>
-    </form>
-    <p>
-       Form Value: {{ recipeForm.value | json }}
-    </p>
\ No newline at end of file
+<div class="container-fluid">
+      <div class="row d-flex justify-content-center">
+         <div class="col-lg-9 col-sm-12">
+            <form [formGroup]="recipeForm" (ngSubmit)="onSubmit()">
+            <div class="form-row">
+               <div class="col">
+                  <h5>Recipe</h5>
+               </div>
+            </div>
+            <div class="form-row form-group">
+               <div class="col">
+                  <label for="name">Name</label>
+                  <input type="text" class="form-control" id="name" formControlName="name" placeholder="Name">
+               </div>
+            </div>
+            <div class="form-row form-group">
+               <div class="col">
+                  <label for="source">Source</label>
+                  <input type="text" class="form-control" id="source" formControlName="source" placeholder="https://www.google.com">
+               </div>
+            </div>
+            <div class="row">
+               <div class="col">
+                  <h5>Time</h5>
+               </div>
+            </div>
+            <div class="form-row form-group">
+               <div class="col">
+                  <label for="prepHour">Prep time</label>
+                  <input type="text" class="form-control" id="prepTime" formControlName="prepTime" placeholder="HH:MM:SS" pattern="((0?[0-9]|[1-9][0-9])(:[0-5][0-9])(:[0-5][0-9]))|(([0-5]?[0-9]):([0-5][0-9]))|([0-5][0-9])">
```

Comment:
This regex is incorrect. It always for inputs such as 0000 which doesn't contain ":" to differentiate between hours, minutes and seconds.

Reply:
^(0[0-9]|0[0-9]|1[0-9]|2[0-3])(:[0-5][0-9]){1,2}$

Reply:
Applied this regex for now.